src/include/stir/recon_buildblock/MLEstimateComponentBasedNormalisation.h

```diff
@@ -0,0 +1,200 @@
+*/
+void ML_estimate_component_based_normalisation(const std::string& out_filename_prefix,
+                                               const ProjData& measured_projdata,
+                                               const ProjData& model_projdata,
+                                               int num_eff_iterations,
+                                               int num_iterations,
+                                               bool do_geo,
+                                               bool do_block,
+                                               bool do_symmetry_per_block,
+                                               bool do_KL,
+                                               bool do_display,
+                                               bool do_save_to_file = true);
+
+/*!
+  \brief Find normalisation factors using a maximum likelihood approach
+  \ingroup recon_buildblock
+*/
+class MLEstimateComponentBasedNormalisation
+{
+public:
+  /*!
+   \brief Constructor
+    \param out_filename_prefix_v The prefix for the output files
+    \param measured_projdata_v The measured projection data
+    \param model_projdata_v The model projection data
+    \param num_eff_iterations_v The number of (sub-)efficiency iterations to perform per iteration of the algorithm
+    \param num_iterations_v The number of algorithm iterations to perform
+    \param do_geo_v Whether to perform geo normalization calculations
+    \param do_block_v Whether to perform block normalization calculations
+    \param do_symmetry_per_block_v Whether to perform block normalization calculations with symmetry
+    \param do_KL_v Whether to perform Kullback-Leibler divergence calculations and display the KL value. This can be slow.
+    \param do_display_v Whether to display the progress of the algorithm.
+    \param do_save_to_file_v Whether to save the each iteration of the efficiencies, geo data and block data to file.
+  */
+  MLEstimateComponentBasedNormalisation(std::string out_filename_prefix_v,
+                                        const ProjData& measured_projdata_v,
+                                        const ProjData& model_projdata_v,
+                                        int num_eff_iterations_v,
+                                        int num_iterations_v,
+                                        bool do_geo_v,
+                                        bool do_block_v,
+                                        bool do_symmetry_per_block_v,
+                                        bool do_KL_v,
+                                        bool do_display_v,
+                                        bool do_save_to_file_v);
+
+  /*!
+  \brief Run the normalisation estimation algorithm using the parameters provided in the constructor.
+  */
+  void process();
+
+  //! Check if the data has been processed
+  bool has_processed_data() const;
```

Comment:
```suggestion
  //! Check if the data has been processed
  bool data_is_processed() const;
```
Also make all setter methods change the returned value to false.